File: SwiftAssignments/SwiftAssignments/Questions/K04.swift
import Foundation

class K04 : Question {
    func question1() {
        print("年齢を入力してください。＞")
        let age = readLine().flatMap { s in Int(s) } ?? 0
        if age < 3 || age >= 70 {
            print("入場料金無料です。")
        } else {
            print("通常料金です。")
        }
    }

    func question2() {
        print("性別を選択してください。（０：男性　１：女性）＞")
        let gender = readLine().flatMap { s in Int(s) } ?? 2
        switch gender {
        case 0:
            print("あら、格好良いですね。")
        case 1:
            print("あら、モデルさんみたいですね。")
        default:
            print("そんな選択肢はありません。")
        }
    }

    func question3() {
        print("年齢を入力してください。＞")
        let age = readLine().flatMap { s in Int(s) } ?? 0
        if age < 0 {
            print("反則です！")
        } else if age < 3 || age >= 70 {
            print("入場料金無料です。")
        } else if age >= 3 && age <= 15 {
            print("子供料金で半額です。")
        } else if age >= 60 && age < 70 {
            print("シニア割引で一割引です。")
        } else {
            print("通常料金です。")
        }
    }

    func question4() {
        print("＊＊＊おみくじプログラム＊＊＊")
        print("おみくじを引きますか　（はい：１　いいえ：０）")
        let choice = readLine().flatMap { s in Int(s) } ?? 0
        if choice < 1 {
            return
        }

        let oracle = Int.random(in: 0...4)
        switch oracle {
        case 0:
            print("大吉　とってもいいことがありそう！")
        case 1:
            print("中吉　きっといいことあるんじゃないかな")
        case 2:
            print("小吉　少しぐらいはいいことあるかもね")
        case 3:
            print("凶　今日はおとなしくておいた方がいいかも")
        case 4:
            print("大凶　これじゃやばくない？早く家に帰った方がいいかも")
        default:
            break
        }
    }

    func execute(number: Int) {
        switch number {
        case 1:
            question1()
        case 2:
            question2()
        case 3:
            question3()
        case 4:
            question4()
        default:
            break
        }
    }


}

File: ScalaRewrite/src/main/scala/assignments/K04.scala
package assignments

import shared.Question

import scala.io.StdIn
import scala.util.Random

case class K04()

object K04 extends Question:
  override def question1(): Unit =
    print("年齢を入力してください。＞")
    val age = StdIn.readInt()
    println(if age < 3 || age >= 70 then "入場料金無料です。" else "通常料金です。")

  override def question2(): Unit =
    print("性別を選択してください。（０：男性　１：女性）＞")
    val gender = StdIn.readInt()
    println(gender match
      case 0 => "あら、格好良いですね。"
      case 1 => "あら、モデルさんみたいですね。"
      case _ => "そんな選択肢はありません。")

  override def question3(): Unit =
    print("年齢を入力してください。＞")
    val age = StdIn.readInt()
    println(getAgeMessage(age))

  override def question4(): Unit =
    println("＊＊＊おみくじプログラム＊＊＊")
    print("おみくじを引きますか　（はい：１　いいえ：０）＞")
    val choice = StdIn.readInt()
    val oracle = Random.between(0, 5)
    println(if choice <= 0 then "" else oracle match
      case 0 => "大吉　とってもいいことがありそう！！"
      case 1 => "中吉　きっといいことあるんじゃないかな"
      case 2 => "小吉　少しぐらいはいいことあるかもね"
      case 3 => "凶　今日はおとなしくておいた方がいいかも"
      case 4 => "大凶　これじゃやばくない？早く家に帰った方がいいかも"
      case _ => "")

  private def getAgeMessage(age: Int): String =
    age match
      case a if a < 3 || a >= 70 => "入場料金無料です。"
      case a if a >= 3 && a <= 15 => "子供料金で半額です。"
      case a if a >= 60 && a < 70 => "シニア割引で一割引きです。"
      case _ => "通常料金です。"